python/lsst/ts/m2/csc.py

```diff
@@ -425,16 +463,18 @@ async def do_enable(self, data):
             salobj.State.OFFLINE, "enterControl", timeout
         )
 
-        # Do the acknowledgement when the controller is in STANDBY state
-        self.cmd_enable.ack_in_progress(data, timeout=timeout)
-
         await self._transition_controller_state(salobj.State.STANDBY, "start", timeout)
         await self._transition_controller_state(
             salobj.State.DISABLED, "enable", timeout
         )
 
         await super().do_enable(data)
 
+    async def begin_disable(self, data: salobj.type_hints.BaseDdsDataType) -> None:
+        self.cmd_disable.ack_in_progress(data, timeout=self.COMMAND_TIMEOUT)
```

Comment:
timeout=3 * self.COMMAND_TIMEOUT. Or specify 1/3 of timeout in do_disable. Or make _transition_controller_state accept time when operation shall be finished - there are 3 _transition_controller_state calls, each can take COMMAND_TIMEOUT at the worst case.